services/webhook/rerun.py

```
@@ -243,6 +346,389 @@ def _run_one(body: str) -> str:
+
+def _stop_review_claim_heartbeat(
+    heartbeat: _ReviewClaimHeartbeat | None,
+) -> bool:
+    if heartbeat is None:
+        return True
+    heartbeat.stop.set()
+    heartbeat.thread.join(timeout=1.0)
+    return not heartbeat.ownership_lost.is_set()
+
+
+def _run_hot_review(
```

Ruff (0.15.20)

[warning] Too many return statements (8 > 6)

(PLR0911)